### PR TITLE
Add improved docs

### PR DESCRIPTION
**Pull request by wooorm**
### Initial checklist

*   [x] I read the support docs 
*   [x] I read the contributing guide 
*   [x] I agree to follow the code of conduct 
*   [x] I searched issues and couldn’t find anything (or linked relevant results below) 
*   [x] If applicable, I’ve added docs and tests

### Description of changes

[View rendered](https://github.com/rehypejs/rehype-sanitize/blob/docs/readme.md)

Related-to: unifiedjs/unified#168.

/cc @nemo-omen @remcohaszing @ChristianMurphy @Murderlon @with-heart @joostdecock @mgan59

### CODE REVIEW

**Review by wooorm — COMMENTED** [review]
✅

**Review by ChristianMurphy — APPROVED** [review]

File: readme.md [hunk 1]
```diff
@@ -79,28 +134,341 @@ The default export is `rehypeSanitize`.
...
+Sanitation schema that defines if and how nodes and properties should be
+cleaned.
 The default schema is exported as `defaultSchema`.
 
+This option is a bit advanced as it requires knowledge of ASTs, so we defer
+to the documentation available for [`Schema` in `hast-util-sanitize`][schema].
+
+## Example
+
+### Example: headings (DOM clobbering)
```
Comment (Murderlon) [inline readme.md:150]: interesting and elaborate example

File: readme.md [hunk 2]
```diff
@@ -79,28 +134,341 @@ The default export is `rehypeSanitize`.
...
++        div: [
++          ...(defaultSchema.attributes.div || []),
++          ['className', 'math', 'math-display']
++        ],
++        span: [
++          ...(defaultSchema.attributes.span || []),
++          ['className', 'math', 'math-inline']
++        ]
++      }
++    })
```
Comment (Murderlon) [inline readme.md:365]: This is a nice example as I comes directly from some issue, if I recall correctly

**Review by Murderlon — APPROVED** [review]

**Review by remcohaszing — APPROVED** [review]

File: readme.md (hunk 1 above)
Comment (wooorm) [reply readme.md:150]: Apparently GH has source maps on their site so it was possible to reverse engineer how they exactly do it. Took some time to remove all the stuff specific to the libraries they use

**Pull request merged**